### PR TITLE
[auth][test] Move request / response types and make tests consistent

### PR DESCRIPTION
**Pull request by jaylees14**
This makes the `auth` service consistent with `user`, as per #47 and #48.

Please compare this with the `user` service directly, we want them to be as close as possible for generation

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]

File: auth/auth_test.go [hunk 1]
```diff
@@ -118,7 +117,7 @@ func TestAuthCreateHandlerSucceeds(t *testing.T) {
 	}
 
 	if id.(float64) != 0 {
-		t.Fatalf("ID is incorrect: found %+v, wanted: 0", id)
+		t.Fatalf("ID is incorrect: found %+v, wanted 0", id)
```
Comment (willjvsmith) [inline auth/auth_test.go]:
```suggestion
		t.Fatalf("ID is incorrect, found: %+v, wanted: 0", id)
```

Comment (willjvsmith) [reply auth/auth_test.go]: and throughout ofc

File: auth/auth_test.go [hunk 2]
```diff
@@ -260,67 +282,86 @@ func TestAuthReadHandlerSucceeds(t *testing.T) {
...
 
-	if res.Code != http.StatusUnauthorized {
+	if res.Code != http.StatusBadRequest {
 		t.Errorf("Wrong status code %v", res.Code)
 	}
 }
 
-// Test that providing an invalid json body causes a `StatusBadRequest`
-func TestAuthReadHandlerFailsOnMalformedJSON(t *testing.T) {
+// Test that providing no body to read endpoint fails
```
Comment (willjvsmith) [inline auth/auth_test.go]:
```suggestion
// Test that providing no body to the read endpoint fails
```

File: auth/auth_test.go [hunk 3]
```diff
@@ -260,67 +282,86 @@ func TestAuthReadHandlerSucceeds(t *testing.T) {
...
-		t.Fatalf("Could not make GET request: %s", err.Error())
+		t.Fatalf("Could not make request: %s", err.Error())
 	}
 
 	if res.Code != http.StatusBadRequest {
+		t.Errorf("Wrong status code: %v", res.Code)
+	}
+}
+
+// Test that providing a non existent auth to the read endpoint fails
```
Comment (willjvsmith) [inline auth/auth_test.go]:
```suggestion
// Test that providing a non-existent auth to the read endpoint fails
```

Comment (willjvsmith) [reply auth/auth_test.go]: and throughout

**Review by willjvsmith — APPROVED** [review]
Few small things, also check comments on #48 and #47

**Pull request merged**